### PR TITLE
Reproject NDCube Documentation

### PR DESCRIPTION
**Pull request by adwaitbhope**
### Description

Added docs for reprojecting `NDCube` (supports #434).

### CODE REVIEW

File: docs/reproject.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. _reproject:
+
+=======================
+Reprojecting ND Objects
+=======================
+
+Reprojecting allows you to transform your ND Objects to use coordinates described by another WCS object.
+Users can use this feature to resample the resolution of ND Objects by providing an appropriate target WCS. It also enables them to remove tiny differences between similar `~ndcube.NDCube` objects to get them onto the same grid.
```
Comment (Cadair) [inline docs/reproject.rst:8]:
```suggestion
Using this feature it is possible to regrid ND Objects by providing an appropriate target WCS, for operations such as resampling or alignment. It also enables removing differences between similar `~ndcube.NDCube` objects to get them onto the same grid.
```

File: docs/reproject.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+.. _cube_reproject:
+
+Reprojecting an NDCube
+======================
+
+Reprojecting returns a new `~ndcube.NDCube` object that has been transformed to use the provided ``target_wcs``.
+The ``target_wcs`` must be compatible with the WCS that is already associated with your `~ndcube.NDCube`. This means that it should represent the same physical axes and in the same order.
```
Comment (Cadair) [inline docs/reproject.rst:16]: Mostly you have done this but rst files should have one sentence per line:

```suggestion
The ``target_wcs`` must be compatible with the WCS that is already associated with your `~ndcube.NDCube`. 
This means that it should represent the same physical axes and in the same order.
```

**Review by Cadair — APPROVED** [review]
Two pretty minor comments, but otherwise looks great :+1:

File: docs/reproject.rst (hunk 2 above)
Comment (adwaitbhope) [reply docs/reproject.rst:16]: Ah okay

File: docs/reproject.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
+.. _reproject:
+
+=======================
+Reprojecting ND Objects
+=======================
+
+Reprojecting allows you to transform your ND Objects to use coordinates described by another WCS object.
+Using this feature it is possible to regrid ND Objects by providing an appropriate target WCS, for operations such as resampling or alignment.
+It also enables removing differences between similar `~ndcube.NDCube` objects to get them onto the same grid.
```
Comment (DanRyanIrish) [inline docs/reproject.rst:9]:
```suggestion
It also enables putting similar `~ndcube.NDCube` objects onto the same grid for more direct comparison.
```

File: docs/reproject.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
+.. _reproject:
+
+=======================
+Reprojecting ND Objects
+=======================
+
+Reprojecting allows you to transform your ND Objects to use coordinates described by another WCS object.
```
Comment (DanRyanIrish) [inline docs/reproject.rst:7]:
```suggestion
Reprojecting allows you to transform your ND Objects to a coordinate grid described by another WCS object.
```

File: docs/reproject.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  >>> target_wcs.wcs.cdelt = 0.1, 0.5, 0.4
+  >>> target_wcs.wcs.crpix = 0, 2, 2
+  >>> target_wcs.wcs.crval = 10, 0.5, 1
+
+  >>> reprojected_cube = my_cube.reproject_to(target_wcs=target_wcs, shape_out=(4, 4, 10))
+
+In the above example, the ``CDELT1`` parameter of the ``target_wcs`` was modified to 0.1 (from 0.2).
+This allowed us to upsample the wavelength axis by a factor of 2.
+Accordingly, the shape of the data was updated to ``(4, 4, 10)`` from the initial shape ``(4, 4, 5)``.
+The wavelength axis goes last since the pixel shape and WCS shape are represented in opposite directions.
```
Comment (DanRyanIrish) [inline docs/reproject.rst:58]:
```suggestion
The wavelength axis goes last since the aray shape and WCS shape are represented in opposite directions.
```

**Review by DanRyanIrish — CHANGES_REQUESTED** [review]
This looks good.  Just a few minor edits required and then it'll be ready to merge.

**Review by Cadair — APPROVED** [review]

**Review by DanRyanIrish — APPROVED** [review]

**Pull request merged**